### ISSUE TITLE
IoT: register_certificate allows the same certificate to be added twice.

### ISSUE THREAD

**Issue by scottbisaillon** [spulec/moto#2273]
Version: 1.3.9

When calling register_certificate on the same certificate twice in a row, the result is that both are added under different certificate ids. This does not match the behavior of the aws service. 

The problem is that a random certificate id is made when a new FakeCertificate is made.

Simple Test Case:
```python
from unittest import TestCase
from moto import mock_iot
import boto3


@mock_iot
class TestRegisterCertificate(TestCase):

    def test_1(self):
        client = boto3.client('iot')
        response = client.register_certificate(certificatePem="some_certificate", setAsActive=False)
        print(response)

        response = client.register_certificate(certificatePem="some_certificate", setAsActive=False)
        print(response)
```

Response:
```bash
{'ResponseMetadata': {'HTTPStatusCode': 200, 'HTTPHeaders': {'server': 'amazon.com'}, 'RetryAttempts': 0}, 'certificateArn': 'arn:aws:iot:us-east-1:1:cert/0b8102cb823dc3249528f9646d84e8904c4f5cf166ec5e5ce3b1e1cc9fa0abb9', 'certificateId': '0b8102cb823dc3249528f9646d84e8904c4f5cf166ec5e5ce3b1e1cc9fa0abb9'}
{'ResponseMetadata': {'HTTPStatusCode': 200, 'HTTPHeaders': {'server': 'amazon.com'}, 'RetryAttempts': 0}, 'certificateArn': 'arn:aws:iot:us-east-1:1:cert/aa67758ed52c424c2ec1a0f1729aca32e5f74cd826a9557775bfba22442a25fa', 'certificateId': 'aa67758ed52c424c2ec1a0f1729aca32e5f74cd826a9557775bfba22442a25fa'}
```

A suggestion would be to base the certificate id off of the certificate_pem that is passed in to ensure the same id is calculated and the correct exception (ResourceAlreadyExistsException) is raised.

**Comment by spulec** [comment]
Thanks for opening. Setting this as a feature request.

**Comment by bpandola** [comment]
Looks like this was implemented in #3331 and should be available in moto >= 1.3.16.dev25

**Comment by bblommers** [comment]
Agreed @bpandola. Will close this - let me know if I've missed something @scottbisaillon
